Editor/Animation/AnimationDatabase.cs

```diff
@@ -171,11 +172,33 @@ internal void RegisterState(AnimatorState state, Action<ClipHolder> processClip
                 {
                     _clipCommitActions.Add(() => { Object.DestroyImmediate(placeholder); });
                 }
+                else
+                {
+                    RegisterMotionMapping(state.motion, placeholder);
+                }
             }
 
             _clipCommitActions.Add(() => { state.motion = clipHolder.CurrentClip; });
         }
 
+        private void RegisterMotionMapping(Motion original, Motion replacement)
+        {
+            ObjectRegistry.RegisterReplacedObject(original, replacement);
+
+            if (original is BlendTree originalTree 
+                && replacement is BlendTree replacementTree 
+                && originalTree.children.Length == replacementTree.children.Length)
```

Comment:
Hmm. It looks like we need to duplicate the contents of the blend tree as well (we're not doing this here, which is probably an oversight...)